harvester_e2e_tests/integration/test_networks.py

```diff
@@ -0,0 +1,919 @@
+                f"Still got {code} with {data}"
+            )
+
+        # Check until VM ip address exists
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+
+        while endtime > datetime.now():
+            code, data = api_client.vms.get_status(unique_name)
+            assert 200 == code, (f"Failed to get specific vm status: {code}, {data}")
+            if 'ipAddress' in data['status']['interfaces'][0]:
+                break
+
+            sleep(5)
+        else:
+            raise AssertionError(
+                f"Failed to get VM {unique_name} IP address, exceed the given timed out\n"
+                f"Still got {code} with {data}"
+            )
+
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        ip_addresses = []
+
+        while endtime > datetime.now():
+            code, data = api_client.vms.get_status(unique_name)
+            assert 200 == code, (f"Failed to get specific vm status: {code}, {data}")
+            if 'interfaces' in data['status']:
+                interfaces_data = data['status']['interfaces']
+                ip_addresses = []
+                if 'ipAddress' in data['status']['interfaces'][0]:
+                    ip_addresses.append(interfaces_data[0]['ipAddress'])
+
+                    if '10.52' in ip_addresses[0]:
+                        break
+                sleep(5)
+        else:
+            raise AssertionError(
+                f"Failed to get VM {unique_name} IP address, exceed the given timed out\n"
+                f"Still got {code} with {data}"
+            )
+
+        # Check can ping management ip address from Harvester node
+        mgmt_ip = ip_addresses[0]
+
+        ping_command = "ping -c 50 {0}".format(mgmt_ip)
+
+        _stdout, _stderr = self.ssh_client(
+            client, vip, node_user, node_password, ping_command, wait_timeout)
+
+        stdout = _stdout.read().decode('ascii').strip("\n")
+
+        assert stdout.find(f"64 bytes from {mgmt_ip}") > 0, (
+            f"Failed to ping VM management IP {mgmt_ip} "
+            f"on management interface from Harvester node")
+
+        # Check can ssh to host and execute command from Harvester node
+        _stdout, _stderr = self.ssh_jumpstart(
+            client, mgmt_ip, vip, node_user, node_password,
+            vm_credential["user"], vm_credential["password"], "ls")
+
+        stdout = _stdout.read().decode('ascii').strip("\n")
+
+        assert stdout.find("bin") == 0, (
+            f"Failed to ssh to VM management IP {mgmt_ip} "
+            f"on management interface from Harvester node")
+
+        # Check should not SSH to management ip address from external host
+        command = ['/usr/bin/ssh', '-o', 'ConnectTimeout=5', mgmt_ip]
+        try:
+            result = subprocess.check_output(
+                command, stderr=subprocess.STDOUT, shell=False, encoding="utf-8")
+        except subprocess.CalledProcessError as e:
```

Comment:
I think we should use `pytest.raises` in here.
https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions

And why we can't use ssh client to test the connection?